File: Python/Assignment2.py
# -*- coding: utf-8 -*-
"""
Created on Mon Jun 29 17:36:32 2015

@author: user
"""
import matplotlib.pyplot as plt
import pandas
import scipy

from Utils import * 

def entries_histogram(turnstile_weather):
    '''
        Entries - Rain vs No-Rain
    '''
    def ifRain(val):
        return turnstile_weather['rain'] == val

    plt.figure()
    plt.suptitle('Histogram - hourly entries (Rain vs No-Rain)')    
    plt.xlabel('Entries')
    plt.ylabel('Frequency')
    
    dataframeForNoRain = turnstile_weather['ENTRIESn_hourly'][ifRain(0)]
    dataframeForNoRain.hist(bins=200)
    dataframeForRain = turnstile_weather['ENTRIESn_hourly'][ifRain(1)]
    dataframeForRain.hist(bins=200)
    
    plt.legend(['No-Rain', 'Rain'])
    plt.axis([0, 5700, 0, 50000])
    
    return plt
 
turnstile_weather = pandas.read_csv(inputFile)   
print entries_histogram(turnstile_weather)

File: R/Assignment2.R
# Question 2
source("util.r")
entries_histogram <- function(turnstile_wether){
  dataFrameForRain <- turnstile_weather[turnstile_weather$rain==0,]
  dataFrameForNoRain <- turnstile_weather[turnstile_weather$rain==1,]
  plot <- hist(dataFrameForRain$ENTRIESn_hourly, col=4, main = "Histogram for Hourly Entries - Rain V/s No rain"
               , xlab="Hourly Entries", xlim=c(0,6000),breaks=200)
  plot2 <- hist(dataFrameForNoRain$ENTRIESn_hourly,col=2,add=T,breaks=200)
  legend("topright", c("No-Rain", "Rain"), fill=c("blue", "red"))
  plot2
}

#turnstile_weather <- read.table(input,header=T,sep=',' )
turnstile_weather <- read.table(file.choose(),header=T,sep=',' )
dim(turnstile_weather)
names(turnstile_weather)
attach(turnstile_weather)

entries_histogram(turnstile_weather)



